### ISSUE TITLE
Wrong code generated for Oracle package procedure SYS.DBMS_STATS.GATHER_DATABASE_STATS

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13625]
When generating code for Oracle package procedure `SYS.DBMS_STATS.GATHER_DATABASE_STATS`, no arguments are generated. It seems there is a mismatch between the reported `OBJECT_ID` values.

This produces `14813`

```sql
select object_id
from all_objects
where object_name = 'DBMS_STATS'
and owner = 'SYS';
```

But this produces `14812`

```sql
select distinct object_id
from all_arguments
where object_name = 'GATHER_DATABASE_STATS';
```

Can be reproduced here: https://dbfiddle.uk/?rdbms=oracle_21&fiddle=43143fc3d8a671293d4809aecaefbd79

As a workaround, we could patch the generated SQL to use names instead of `OBJECT_ID`...

**Comment by lukaseder** [comment]
I think we need to work with the `OBJECT_ID` rather than just the name to make sure we're not accidentally duplicating things in the presence of a `SYNONYM` by the same name.

**Comment by lukaseder** [comment]
Well, this query here produces way more results (hundreds) than I would have expected (0):

```sql
select object_id
from all_procedures
except
select object_id
from all_objects
where object_type in (
  'FUNCTION',
  'PACKAGE',
  'PROCEDURE',
  'TYPE',
  'TRIGGER'
);
```

**Comment by lukaseder** [comment]
Our integration tests suggest that we can omit the `OBJECT_ID` predicate in `OracleRoutineDefinition::init0`:

```java
//  .and(a.OBJECT_ID.equal(objectId))
```

If we make sure to add a predicate of the form:

```java
.and(getPackage() == null ? noCondition() : a.PACKAGE_NAME.equal(getPackage().getInputName()))
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.7 (https://github.com/jOOQ/jOOQ/issues/13626), 3.15.11 (https://github.com/jOOQ/jOOQ/issues/13627), and 3.14.16 (https://github.com/jOOQ/jOOQ/issues/13628)
